moderation/commands.go

```diff
@@ -1015,12 +1015,27 @@ var ModerationCommands = []*commands.YAGCommand{
 				return nil, err
 			}
 
-			_, err = MBaseCmdSecond(parsed, "", true, discordgo.PermissionManageMessages, config.WarnCmdRoles, config.WarnCommandsEnabled, true)
+			reason := SafeArgString(parsed, 1)
+			reason, err = MBaseCmdSecond(parsed, reason, true, discordgo.PermissionManageMessages, config.WarnCmdRoles, config.WarnCommandsEnabled, true)
 			if err != nil {
 				return nil, err
 			}
 
 			warningID := parsed.Args[0].Int()
+
+			// Using that if we need to send to modlog
+			var warning *models.ModerationWarning
+			if config.DelwarnSendToModlog && config.ActionChannel != 0 {
+				warning, err = models.ModerationWarnings(
+					models.ModerationWarningWhere.ID.EQ(warningID),
+					// don't get warning from other servers, even if ID is correct
+					models.ModerationWarningWhere.GuildID.EQ(parsed.GuildData.GS.ID),
+				).OneG(parsed.Context())
+				if err != nil {
+					return fmt.Sprintf("Could not find warning with ID `%d`", warningID), nil
+				}
+			}
+
 			numDeleted, err := models.ModerationWarnings(
 				models.ModerationWarningWhere.ID.EQ(warningID),
```

Comment:
I think fetching `warning` unconditionally and calling `warning.Delete` is more straightforward here. You can remove the separate deletion query below if you do this.